### PR TITLE
tests: 🧪 add cases to test `classify_diabetes()`

### LINKED ISSUE #281: Start filling in explicit tests in `test-classify-diabetes.R`
See for example `test-prepare-lpr.R` file.

So that we know for certain which IDs should be correctly classified given a specific input data in the registers.

### LINKED ISSUE #303: Create fake case data set that contains potentially real cases/edge cases
name like `fake_cases`? 🤷

### PR DESCRIPTION
**Pull request by Aastedet**
## Description

- Added `edge-cases.R` functions to generate dataset of 23 edge cases and expected outputs.
- Added tests for `classify_diabetes()`to `test-classify-diabetes.R` . The default outputs from `testthat::expect_identical()` seem sufficient for what we need, so I scrapped the pnr-by-pnr testing loop from the previous PR.
-  Fixes for a few bugs in `classify_diabetes()` and `create_inclusion_dates()`
- Removed the old, commented-out tests in `test-classify-diabetes.R`
- Case description vignette is NOT here - has been split into separate PR.

Closes #303, closes #281 

## Checklist

- [ ] Ran `just run-all`
- [ ] If docs were added, Markdown is formatted
- [x] Nuke the previous PR and never mention it again.

### CODE REVIEW

File: R/edge-cases.R [hunk 1]
```diff
@@ -0,0 +1,384 @@
...
+#'
+#' The generated data is used in `testthat` tests to ensure the
+#' algorithm behaves as expected under a wide range of conditions, but it is also intended to be explored by users to better understand how the algorithm logic works.
+#'
+#' @return A [tibble::tibble()] with the following columns:
+#'   - `pnr`: The personal identification variable.
+#'   - `stable_inclusion_date`: The expected stable inclusion date.
+#'   - `raw_inclusion_date`: The expected raw inclusion date (date of the
+#'     second-ever inclusion event).
+#'   - `diabetes_type`: The expected diabetes type ("T1D" or "T2D").
```
Comment (signekb) [inline R/edge-cases.R:350]: According to #344, the expected output now contains `has_t1d` and `has_t2d` instead. 
See the [Output](https://steno-aarhus.github.io/osdc/articles/design.html#output-1) section in the Design vignette.

File: R/edge-cases.R [hunk 2]
```diff
@@ -0,0 +1,384 @@
...
+#' @export
+#' @seealso See the [edge-case-description] vignette for a detailed description of these cases.
+#'
+#' @examples
+#' \dontrun{
+#' expected_inclusions <- create_expected_inclusions()
+#' }
+create_expected_inclusions <- function() {
+  tibble::tribble(
+    ~pnr,                                     ~stable_inclusion_date, ~raw_inclusion_date, ~diabetes_type,
```
Comment (signekb) [inline R/edge-cases.R:361]:
```suggestion
    ~pnr,                                     ~stable_inclusion_date, ~raw_inclusion_date, ~has_t1d, ~has_t2d,
```

File: tests/testthat/test-classify-diabetes.R [hunk 3]
```diff
@@ -1,254 +1,30 @@
...
-#   1, "2000-01-02", NA, NA, TRUE, TRUE,
-#   1, "2015-01-02", NA, NA, TRUE, TRUE,
-#   3, "2010-02-02", NA, NA, TRUE, TRUE
-# ) |>
-#   dplyr::mutate(date = lubridate::as_date(date))
+test_that("Algorithm produced unexpected outputs", {
+  # Test inputs:
+  edge_case_data <- create_test_cases()
+
+  # Generate actual outputs:
```
Comment (signekb) [inline tests/testthat/test-classify-diabetes.R:5]:
```suggestion
```

File: tests/testthat/test-classify-diabetes.R [hunk 4]
```diff
@@ -1,254 +1,30 @@
...
-#   2, "2010-02-02", TRUE, NA, TRUE,
-#   # From included_hba1c.
-#   1, "2000-01-01", NA, NA, TRUE, TRUE, # Same pnr and date as row from excluded_pcos, both kept.
-#   1, "2000-01-02", NA, NA, TRUE, TRUE,
-#   1, "2015-01-02", NA, NA, TRUE, TRUE,
-#   3, "2010-02-02", NA, NA, TRUE, TRUE
-# ) |>
-#   dplyr::mutate(date = lubridate::as_date(date))
+test_that("Algorithm produced unexpected outputs", {
+  # Test inputs:
```
Comment (signekb) [inline tests/testthat/test-classify-diabetes.R:2]:
```suggestion
```
Not needed :)

File: tests/testthat/test-classify-diabetes.R [hunk 5]
```diff
@@ -1,254 +1,30 @@
...
+    lpr_diag = edge_case_data$lpr_diag,
+    lpr_adm = edge_case_data$lpr_adm,
+    sysi = edge_case_data$sysi,
+    sssy = edge_case_data$sssy,
+    lab_forsker = edge_case_data$lab_forsker,
+    bef = edge_case_data$bef,
+    lmdb = edge_case_data$lmdb
+  )
+
+  # Specify expected outputs (included cases):
```
Comment (signekb) [inline tests/testthat/test-classify-diabetes.R:18]:
```suggestion
```

**Review by signekb — CHANGES_REQUESTED** [review]
Very nice work, Anders! 🎉 
I have added some suggestions below:

File: R/edge-cases.R [hunk 6]
```diff
@@ -0,0 +1,384 @@
...
+#'   `kontakter`, `diagnoser`, `sysi`, `sssy`, and `lab_forsker`.
+#' @export
+#' @seealso See the [edge-case-description]  vignette for a detailed description of these cases.
+#'
+#' @examples
+#' \dontrun{
+#' edge_case_input_data <- create_test_cases()
+#' }
+create_test_cases <- function() {
+  # 1. bef: Demographics table -------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (Aastedet) [reply R/edge-cases.R:23]: Good idea. I'll remove these ones though - they were mainly there as quality of life improvements during the reviewing process (less scrolling from register to register).

File: R/edge-cases.R (hunk 1 above)
Comment (Aastedet) [reply R/edge-cases.R:350]: Thanks! 👀 I didn't catch that.
The Vignette table columns are called `has_t1d` and `has_t1d` though, so I'll edit that file as well when I fix this.
